### PR TITLE
Ranking: standardize symbol scores

### PR DESCRIPTION
**Pull request by jtibshirani**
I found this code hard to follow and pulled out an enum to help unify things (shown in this PR). Then I realized the language-specific scoring sometimes uses custom factor values that don't match the default ones.

So just opening a draft to get feedback on a proposal. I'd like to use the same set of "symbol kinds" and scores for all languages. The only language-specific differences would be capturing more ctags labels (like "classes" instead of just "class", handling PHP "traits", etc.) This will simplify the logic and make it easier to reason about, since the same kind of symbol will always contribute to the score in the same way. The change would affect scoring, but I don't think the current behavior was so carefully chosen that it'd have a bad effect.

### CODE REVIEW

File: contentprovider.go [hunk 1]
```diff
@@ -694,64 +709,64 @@ func scoreKind(language string, kind string) float64 {
...
 		switch kind {
 		case "class":
-			factor = 10
+			score = Class
 		case "interface":
-			factor = 9
+			score = Enum
 		case "method":
-			factor = 8
+			score = Interface
```
Comment (stefanhengl) [inline contentprovider.go:731]: This confuses me more than before. I think it is ok that EG "interface" has a different score depending on the language. Interface might be a super important in one language, like Go, and a tiny bit less important in class based languages.

Comment (stefanhengl) [reply contentprovider.go:731]: I'd rather replace the strings with an enum and define mapping for both universal-ctags and scip-ctags to our internal representation.

Comment (jtibshirani) [reply contentprovider.go:731]: > This confuses me more than before. I think it is ok that EG "interface" has a different score depending on the language.

In case it wasn't clear, my proposal was to change this so that interfaces, etc. are treated the same across languages. Then this refactor will be clean and not so confusing :)

Anyways thanks for your thoughts, let's discuss more in our next meeting!

Comment (jtibshirani) [reply contentprovider.go:731]: Now that I'm up-to-speed on how this works, I understand and think we need a different approach. Closing this out.
